src/main/java/io/vertx/openapi/validation/impl/RequestValidatorImpl.java

```diff
@@ -153,6 +154,28 @@ RequestParameter validateBody(RequestBody requestBody, ValidatableRequest reques
       throw new ValidatorException("The format of the request body is not supported", UNSUPPORTED_VALUE_FORMAT);
     }
     Object transformedValue = transformer.transformRequest(mediaType, request);
+    // Corner case for "application/octet-stream" media type:
+    // Skip full schema validation if the following simplified validation succeeds,
+    // otherwise validate with JSON Validator and fail the validation
+    // of the transformed value eventually.
+    if (mediaType.getIdentifier().equals(MediaType.APPLICATION_OCTET_STREAM)) {
+      JsonSchema schema = mediaType.getSchema();
+      String schemaTypeValue = schema.get("type", null);
+      if (schemaTypeValue != null && schemaTypeValue.equals("string")) {
+        String schemaFormatValue = schema.get("format", null);
```

Comment:
for `format` it's the same as for `type`. If there is a chance that this required information is missing, it should fail when building the contract object, not when we do a validation.

Reply:
`format` can be `null`, since OpenAPI 3.1 replaced it with `contentMediaType`. I'm merely providing a fallback if someone still uses `format` in the contract file.

Reply:
The simplified validation procedure has to deal with both 3.0 and 3.1 of OpenAPI. Therefore the statements used seem to be redundant.

Reply:
Okay, you are right, format can be null. But as mentioned in the comment below, I'm not convinced by the current implementation of the "schema validation bypass".

My concerns:
 - `RequestValidatorImpl` is already a large class
 - different logic for different versions -> makes it complex
 - you have to take care for corner cases

But I understand the benefit of having a schema validation bypass, especially for binary data.

Maybe we can call a method like "canSkipValidation" and extract this method and its logic into a separated class. You could pass the contract reference to this method to check if the contract is 3.0 or 3.1 which makes the logic in this method less complex.

What do you think?